RealmSwift/Tests/RealmTests.swift

```diff
@@ -0,0 +1,475 @@
+            XCTAssertNil(error)
+        }
+
+        NSFileManager.defaultManager().createFileAtPath(Realm.defaultPath,
+            contents:"a".dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: false),
+            attributes: nil)
+        XCTAssertNil(Realm(path: Realm.defaultPath, readOnly: false, error: &error), "Should not throw with error")
+        XCTAssertNotNil(error)
+
+        assertThrows(Realm(path: Realm.defaultPath, readOnly: false, error: nil))
+        assertThrows(Realm(path: Realm.defaultPath, readOnly: false))
+        assertThrows(Realm(path: Realm.defaultPath, readOnly: false, encryptionKey: "asdf".dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: false), error: &error))
+        assertThrows(Realm(path: "", readOnly: false, error: &error))
+    }
+
+    func testInitInMemory() {
+        autoreleasepool {
+            var realm = Realm(inMemoryIdentifier: "identifier")
+            realm.write {
+                realm.create(SwiftIntObject.self, value: [1])
+                return
+            }
+        }
+        var realm = Realm(inMemoryIdentifier: "identifier")
+        XCTAssertEqual(realm.objects(SwiftIntObject).count, 0)
+
+        realm.write {
+            realm.create(SwiftIntObject.self, value: [1])
+            XCTAssertEqual(realm.objects(SwiftIntObject).count, 1)
+
+            Realm(inMemoryIdentifier: "identifier").create(SwiftIntObject.self, value: [1])
+            XCTAssertEqual(realm.objects(SwiftIntObject).count, 2)
+        }
+
+        var realm2 = Realm(inMemoryIdentifier: "identifier2")
+        XCTAssertEqual(realm2.objects(SwiftIntObject).count, 0)
+    }
+
+    func testWrite() {
+        Realm().write {
+            self.assertThrows(Realm().beginWrite())
+            self.assertThrows(Realm().write { })
+            Realm().create(SwiftStringObject.self, value:["1"])
+            XCTAssertEqual(Realm().objects(SwiftStringObject).count, 1)
+        }
+        XCTAssertEqual(Realm().objects(SwiftStringObject).count, 1)
+    }
+
+    func testBeginWrite() {
+        Realm().beginWrite()
+        assertThrows(Realm().beginWrite())
+        Realm().cancelWrite()
+        Realm().beginWrite()
+        Realm().create(SwiftStringObject.self, value:["1"])
+        XCTAssertEqual(Realm().objects(SwiftStringObject).count, 1)
+    }
+
+    func testCommitWrite() {
+        Realm().beginWrite()
+        Realm().create(SwiftStringObject.self, value:["1"])
+        Realm().commitWrite()
+        XCTAssertEqual(Realm().objects(SwiftStringObject).count, 1)
+        Realm().beginWrite()
+    }
+
+    func testCancelWrite() {
+        assertThrows(Realm().cancelWrite())
+        Realm().beginWrite()
+        Realm().create(SwiftStringObject.self, value:["1"])
+        Realm().cancelWrite()
+        XCTAssertEqual(Realm().objects(SwiftStringObject).count, 0)
+
+        Realm().write {
+            self.assertThrows(self.realmWithTestPath().cancelWrite())
```

Comment:
should insert an object and check for zero afterwards again to verify it actually cancels the block write and doesn't just fail to throw